src/py/flwr/server/run_serverapp.py

```diff
@@ -136,32 +137,50 @@ def run_server_app() -> None:
             cert_path,
         )
 
-    log(
-        DEBUG,
-        "Flower will load ServerApp `%s`",
-        getattr(args, "server-app"),
+    if not (getattr(args, "server-app") is None) ^ (args.run_id is None):
+        raise sys.exit(
+            "Invalid options: Either `server-app` or `--run-id` "
+            "should be set but not both."
+        )
+
+    stub = GrpcDriverStub(
+        driver_service_address=args.superlink, root_certificates=root_certificates
     )
+    # Create run if run_id is not provided
+    if args.run_id is None:
+        stub.connect()
+        req = CreateRunRequest(fab_id=args.fab_id, fab_version=args.fab_version)
+        res = stub.create_run(req)
+        run_id = res.run_id
+    else:
+        run_id = args.run_id
+
+    # Initialize GrpcDriver
+    driver = GrpcDriver(run_id=run_id, stub=stub)
+
+    # Dynamically obtain ServerApp path based on run_id
+    if args.run_id is not None:
+        if args.flwr_dir is None:
+            flwr_dir = get_flwr_dir()
+        else:
+            flwr_dir = Path(args.flwr_dir).absolute()
```

Comment:
nit: we could move this if/else into `get_flwr_dir()` so that getting the flwr dir becomes a consistent one-liner everywhere:
```suggestion
        flwr_dir = get_flwr_dir(args.flwr_dir)
```

Reply:
Here it is: #3661

Reply:
Thanks, merged